EmbodiedSuffering_Engine/Compute/SufferingIndex.cs

```diff
@@ -0,0 +1,115 @@
+            List<string> invalidCountries = new List<string>();
+
+            // List length check: Fail the calculation if the lists are of unequal lengths
+            if (numberOfEnslavedPeople.Count != importRatios.Count)
+            {
+                int numCount = numberOfEnslavedPeople.Count;
+                int ratiosCount = importRatios.Count;
+                BH.Engine.Base.Compute.RecordError("The provided list's lengths to do correspond as expected." + $"\nNumberOfEnslavedPeople count = {numCount}" + $"\nImportRatios count = {ratiosCount}. \nPlease update your data and try again.");
+            } else
+            {
+                // Remove after debugging is complete
+                BH.Engine.Base.Compute.RecordNote("Lists are the same length, proceeding to threshold check.");
+            }
+
+            // Add countries to ImportCountries[]
+            for (int i = 0; i < materialImportSources.ImportRatios.Count; i++)
+            {
+                exportCountries.Add(materialImportSources.ExportCountries);
+            }
+
+            // Pass a list of string import countries
+            // This is used in the threshold check
+            List<string> importCountriesAsStrings = exportCountries.ConvertAll(x => x.ToString());
+
+            // Threshold check: Toss anything above the input threshold into badCountries[]
+            for (int i = 0; i < numberOfEnslavedPeople.Count; i++)
+            {
+                if (numberOfEnslavedPeople[i] > acceptableThreshold)
+                {
+                    culledCountries.Add(importCountriesAsStrings[i]);
+                    numberOfEnslavedPeople.Remove(numberOfEnslavedPeople[i]);
+                    BH.Engine.Base.Compute.RecordWarning("Some countries were removed from your calculation due to exceeding the acceptableThreshold requirement. \nPlease review them from the CulledCountries output. \nThe calculation will now proceed with all remaining countries.");
+                }
+            }
+
+            // Data validitiy check: Toss anything with 0 or NaN into invalidCountries[]
+            for (int i = 0; i < numberOfEnslavedPeople.Count; i++)
+            {
+                if (numberOfEnslavedPeople[i] == double.NaN)
+                {
+                    numberOfEnslavedPeople[i] = 0;
+                    invalidCountries.Add(importCountriesAsStrings[i]);
+                    BH.Engine.Base.Compute.RecordWarning("Some country's number of enslaved people was defaulted to zero due to missing or invalid data. \nPlease review them from the invalidCountries output. \nThe calculation will now proceed with all remaining countries.");
+                }
+            }
+
+            // Calculation: Multiply the lists of number of people by the import ratios
+            for (int i = 0; i < numberOfEnslavedPeople.Count; i++)
+            {
+                sufferingIndex += (numberOfEnslavedPeople[i] * materialImportSources.ImportRatios[i]);
+            } 
```

Comment:
@kayleighhoude this is the math I derived from the issue. I'm not dividing it by the list length, but I can do that.